packages/radix-vue/src/Tooltip/TooltipContent.vue

```diff
@@ -1,38 +1,116 @@
+   * content that cannot be announced, use aria-label as a more
+   * descriptive label.
+   *
+   * @default String
+   */
+  ariaLabel?: string;
+}
+
+export interface TooltipContentEmit {
+  (e: "escapeKeyDown", event: KeyboardEvent): void;
+  (e: "pointerDownOutside", event: Event): void;
 }
 </script>
 
 <script setup lang="ts">
-import { inject } from "vue";
+import { PopperContent } from "@/Popper";
 import {
-  TOOLTIP_INJECTION_KEY,
-  type TooltipProvideValue,
-} from "./TooltipRoot.vue";
-import { PrimitiveDiv } from "@/Primitive";
-import { PopperContent, type PopperContentProps } from "@/Popper";
+  PrimitiveDiv,
+  ScreenReaderOnly,
+  type PrimitiveProps,
+} from "@/Primitive";
+import { computed, inject, ref, useSlots, type VNode } from "vue";
+import { TOOLTIP_INJECTION_KEY } from "./TooltipRoot.vue";
+
+const contentElement = ref<HTMLElement>();
 
-const injectedValue = inject<TooltipProvideValue>(TOOLTIP_INJECTION_KEY);
+const injectedValue = inject(TOOLTIP_INJECTION_KEY);
 
 const props = withDefaults(defineProps<TooltipContentProps>(), {
+  asChild: false,
   side: "top",
+  sideOffset: 0,
   align: "center",
+  avoidCollisions: true,
+  collisionBoundary: () => [],
+  collisionPadding: 0,
+  arrowPadding: 0,
+  sticky: "partial",
+  hideWhenDetached: false,
+});
+
+const emit = defineEmits<TooltipContentEmit>();
+
+onClickOutside(contentElement, () => {
+  emit("pointerDownOutside", new Event("pointerdown"));
+});
+
+function onEscapeKeyDown(event: KeyboardEvent) {
+  emit("escapeKeyDown", event);
+}
+
+const ariaLabel = computed(() => {
+  if (props.ariaLabel) return props.ariaLabel;
+  const defaultSlot = useSlots().default?.();
+  let content: string = "";
+
+  function recursiveTextSearch(node: VNode) {
+    if (typeof node.children === "string") {
+      content += node.children;
+    } else if (Array.isArray(node.children)) {
+      node.children.forEach((child) => recursiveTextSearch(child as VNode));
+    }
+  }
+
+  defaultSlot?.forEach((node) => recursiveTextSearch(node));
+  return content;
 });
 </script>
 
 <template>
-  <PopperContent v-bind="props" v-if="injectedValue?.open.value">
-    <PrimitiveDiv
-      ref="primitiveElement"
-      :data-state="injectedValue.open.value ? 'delayed-open' : 'closed'"
-      :data-side="side"
-      :data-align="align"
-      role="tooltip"
-      tabindex="-1"
-      :asChild="props.asChild"
-    >
+<PopperContent v-if="injectedValue?.open.value" ref="contentElement" :as-child="props.asChild" :side="props.side"
+    :sideOffset="props.sideOffset" :align="props.align" :alignOffset="props.alignOffset"
+    :avoidCollisions="props.avoidCollisions" :collisionBoundary="props.collisionBoundary"
+    :collisionPadding="props.collisionPadding" :arrowPadding="props.arrowPadding" :sticky="props.sticky"
+    :hideWhenDetached="props.hideWhenDetached" style="
+      --radix-tooltip-content-transform-origin: var(
+        --radix-popper-transform-origin
+      );
```

Comment:
what's the reason replacing `v-bind=props` with explicitly props ya? It kinda congested the code, and serves no different as v-bind props.

Reply:
Basically to avoid `props` drilling; Avoid passing to `PopperContent` other variables that they don't use like `ariaLabel`.
For sure, I would prefer using `v-bind` as it is concise, but I just did this to be safe